### PR TITLE
amp-ad-exit: ignore clicks on specific elements

### PR DESCRIPTION
**Pull request by clawr**
Add a new type of filter to amp-ad-exit to ignore clicks on certain elements specified by the document author. For example, if the entire creative should be clickable except for an [x] button in the corner, the exit action could be on the body with an inactiveElement filter that matches the close button.

An instance of this filter is added by default to amp-ad-exit to ignore clicks on the slide navigation buttons of `<amp-carousel>` (elements with class `amp-carousel-button`).

### CODE REVIEW

**Comment by jonkeller** [comment]
Sorry Curt, was traveling back from Amsterdam. Will review this afternoon.

File: examples/amp-ad-exit.amp.html [hunk 1]
```diff
@@ -88,5 +97,16 @@ <h1 on="tap:exit-api.exit(target='target1')" role="button" tabindex="1">amp-ad-e
   </div>
   <p><a href="https://www.google.com/" on="tap:exit-api.exit(target='target1', _elem='anchor')">&lt;A&gt; tags don't work well (try middle-clicking)</a>
 </div>
+
+<div class="ad" on="tap:exit-api.exit(target='target1')" role="button" tabindex="1">
```
Comment (jonkeller) [inline examples/amp-ad-exit.amp.html]: Nit: Tabindex 1 already in use on line 91

File: extensions/amp-ad-exit/amp-ad-exit.md [hunk 2]
```diff
@@ -165,6 +165,23 @@ The `clickDelay` filter type specifies the time to wait before responding to cli
   </tr>
 </table>
 
+### inactiveElement filter
+
+The `inactiveElement` filter type specifies elements that should not cause exits when they are the source of an event. The `amp-ad-exit` element ignores clicks on the previous/next buttons of an `amp-carousel` by default. The `inactiveElement` filter requires the following properties:
```
Comment (jonkeller) [inline extensions/amp-ad-exit/amp-ad-exit.md]: This is super cool!

File: examples/amp-ad-exit.amp.html (hunk 1 above)
Comment (clawr) [reply examples/amp-ad-exit.amp.html]: tabindex doesn't need to be unique - it's more like a class than id
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

File: extensions/amp-ad-exit/0.1/filters/inactive-element.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Filter, FilterType} from './filter';
+import {matches} from '../../../../src/dom';
+import {user} from '../../../../src/log';
+
+/**
+ * A Filter that ignores events originating from elements that match the spec.
```
Comment (lannka) [inline extensions/amp-ad-exit/0.1/filters/inactive-element.js]: match an element selector

Comment (clawr) [reply extensions/amp-ad-exit/0.1/filters/inactive-element.js]: Done

**Comment by clawr** [comment]
I think this is ready to merge. The two failing tests are not for amp-ad-exit.

**Comment by jonkeller** [comment]
The failed tests were just Sauce Labs timeouts, which are all too common. I restarted them - hopefully they'll pass this time.

**Comment by clawr** [comment]
Is anything blocking the merge?

**Comment by jonkeller** [comment]
Nothing blocking merge on my end. I do not have permissions to do the merge myself.

**Pull request merged**